vm/proxyapp/proxyappclient.go

```diff
@@ -151,19 +206,16 @@ func runProxyApp(params *proxyAppParams, cmd string) (*ProxyApp, error) {
 		cancelContext()
 	}
 
-	subStdout, err := subProcess.StdoutPipe()
-	if err != nil {
-		freeAll()
-		return nil, fmt.Errorf("failed to get stdoutpipe: %w", err)
-	}
-	toClose = append(toClose, subStdout)
```

Comment:
Could you please clarify your concern?
I don't call subProcess.StdoutPipe() here. I don't understand what could leak.

Reply:
Currently we call `subProcess.StdoutPipe()` and then add the pipe to the `toClose` slice.
If later `subProcess.StderrPipe()` or `subProcess.Start()` fails, we call `freeAll` and close stdout pipe.

After this change, due to the accompanying refactoring, we don't add stdout/stdin pipes to the `toClose` slice. So if `subProcess.StderrPipe()` or `subProcess.Start()` fail later, we won't close them.
Now we add `client` to the `toClose` slice, but since it's constructed using `io.NopCloser`, it won't close anything (adding it to `toClose` is pointless).

Is this change in behavior intentional?

Reply:
I've added one more closer. Thanks.